In [7]:
import csv

file = open("guns.csv", "r")
guns_wo_header = list(csv.reader(file))[1:]
guns_first_10 = guns_wo_header[:10]

print(guns_first_10)
print(len(guns_wo_header))

[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', 'BA+'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', 'Some college'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', 'BA+'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', 'BA+'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', 'HS/GED'], ['6', '2012', '02', 'Suicide', '0', 'M', '17', 'Native American/Native Alaskan', '100', 'Home', 'Less than HS'], ['7', '2012', '02', 'Undetermined', '0', 'M', '48', 'White', '100', 'Home', 'HS/GED'], ['8', '2012', '03', 'Suicide', '0', 'M', '41', 'Native American/Native Alaskan', '100', 'Home', 'HS/GED'], ['9', '2012', '02', 'Accidental', '0', 'M', '50', 'White', '100', 'Other specified', 'Some college'], ['10', '2012', '02', 'Suicide', '0', 'M', 'NA', 'Black', '998', 'Home', 'NA']]
100798


In [8]:
years = [gun[1] for gun in guns_wo_header]
print(years[:10])
print(len(years))

['2012', '2012', '2012', '2012', '2012', '2012', '2012', '2012', '2012', '2012']
100798


In [9]:
year_counts = {}

for year in years:
    if year in year_counts:
        year_counts[year] += 1
    else:
        year_counts[year] = 1
        
print(year_counts)

{'2012': 33563, '2013': 33636, '2014': 33599}


It looks like gun deaths didn't change much by year from 2012 to 2014. Let's see if gun deaths in the US change by month and year. In order to do this, we'll have to create a datetime.datetime object using the year and month columns. We'll then be about to count up gun deaths by date, like we did by year in the last screen.

In [16]:
import datetime

dates = [datetime.datetime(year=int(row[1]),month=int(row[2]),day=15) for row in guns_wo_header]
print(dates[:10])

[datetime.datetime(2012, 1, 15, 0, 0), datetime.datetime(2012, 1, 15, 0, 0), datetime.datetime(2012, 1, 15, 0, 0), datetime.datetime(2012, 2, 15, 0, 0), datetime.datetime(2012, 2, 15, 0, 0), datetime.datetime(2012, 2, 15, 0, 0), datetime.datetime(2012, 2, 15, 0, 0), datetime.datetime(2012, 3, 15, 0, 0), datetime.datetime(2012, 2, 15, 0, 0), datetime.datetime(2012, 2, 15, 0, 0)]


In [17]:
dates_count = {}

for date in dates:
    if date in dates_count:
        dates_count[date] += 1
    else:
        dates_count[date] = 1
        
print(dates_count)

{datetime.datetime(2012, 1, 15, 0, 0): 2758, datetime.datetime(2012, 2, 15, 0, 0): 2357, datetime.datetime(2012, 3, 15, 0, 0): 2743, datetime.datetime(2012, 4, 15, 0, 0): 2795, datetime.datetime(2012, 5, 15, 0, 0): 2999, datetime.datetime(2012, 6, 15, 0, 0): 2826, datetime.datetime(2012, 7, 15, 0, 0): 3026, datetime.datetime(2012, 8, 15, 0, 0): 2954, datetime.datetime(2012, 9, 15, 0, 0): 2852, datetime.datetime(2012, 10, 15, 0, 0): 2733, datetime.datetime(2012, 11, 15, 0, 0): 2729, datetime.datetime(2012, 12, 15, 0, 0): 2791, datetime.datetime(2013, 1, 15, 0, 0): 2864, datetime.datetime(2013, 2, 15, 0, 0): 2375, datetime.datetime(2013, 3, 15, 0, 0): 2862, datetime.datetime(2013, 4, 15, 0, 0): 2798, datetime.datetime(2013, 5, 15, 0, 0): 2806, datetime.datetime(2013, 6, 15, 0, 0): 2920, datetime.datetime(2013, 7, 15, 0, 0): 3079, datetime.datetime(2013, 8, 15, 0, 0): 2859, datetime.datetime(2013, 9, 15, 0, 0): 2742, datetime.datetime(2013, 10, 15, 0, 0): 2808, datetime.datetime(2013, 11,

The sex and race columns contain potentially interesting information on how gun deaths in the US vary by gender and race. Exploring both of these columns can be done with a similar dictionary counting technique to what we did earlier.

In [23]:
#sex
sexes = [row[5] for row in guns_wo_header]

sex_count = {}
for sex in sexes:
    if sex in sex_count:
        sex_count[sex] += 1
    else:
        sex_count[sex] = 1
        
print(sex_count)

#relative amounts
sex_count_rel = {}
for sex in sex_count:
    sex_count_rel[sex] = sex_count[sex]/len(sexes)

print(sex_count_rel)


#race
races = [row[7] for row in guns_wo_header]

race_count = {}
for race in races:
    if race in race_count:
        race_count[race] += 1
    else:
        race_count[race] = 1
        
print(race_count)

#relative amounts
race_count_rel = {}
for race in race_count:
    race_count_rel[race] = race_count[race]/len(races)

print(race_count_rel)




{'M': 86349, 'F': 14449}
{'M': 0.8566539018631322, 'F': 0.1433460981368678}
{'Asian/Pacific Islander': 1326, 'White': 66237, 'Native American/Native Alaskan': 917, 'Black': 23296, 'Hispanic': 9022}
{'Asian/Pacific Islander': 0.01315502291712137, 'White': 0.6571261334550289, 'Native American/Native Alaskan': 0.009097402726244569, 'Black': 0.23111569674001467, 'Hispanic': 0.0895057441615905}


# Findings

## Sex
* 85.7% of deaths are male
* 14.3% of deaths are female

## Race
* 1.3% of deaths are Asian / Pacific Islander
* 65.7% of deaths are White
* 0.1% of deaths are NA/NA
* 23.1% of deaths are Black
* 0.9% of deaths are Hispanic

--> Majority is **male and white**


# Interesting to know
* how are deaths distributed among months (e.g. more in winter?)
* how are deaths distributed among locations (home, street, ...)
* whats the diatribution pf education level among deaths?

## Add census data

We explored gun deaths by race in the past screen. However, our analysis only gives us the total number of gun deaths by race in the US. Unless we know the proportion of each race in the US, we won't be able to meaningfully compare those numbers. What we really want to get is a rate of gun deaths per 100000 people of each race. In order to do this, we'll need to read in data about what percentage of the US population falls into each racial category. Luckily, we can import some census data to help us out.

The data contains information on the total population of the US, as well as the total population of each racial group in the US. The data is stored in the census.csv file, and only consists of two rows:

In [24]:
census = list(csv.reader(open("census.csv","r")))
print(census)

[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


In [39]:

mapping = {
    "Asian/Pacific Islander" : (census[1][14] + census[1][15]), 
    "Black": census[1][12], 
    "Native American/Native Alaskan" : census[1][13] , 
    "Hispanic": census[1][11],
    "White" : census[1][10]  
}

print(mapping)
    

mapping_rel = {
    "Asian/Pacific Islander" : ((int(census[1][14]) + int(census[1][15]))/int(census[1][9])), 
    "Black": int(census[1][12])/int(census[1][9]), 
    "Native American/Native Alaskan" : int(census[1][13])/int(census[1][9]), 
    "Hispanic": int(census[1][11])/int(census[1][9]),
    "White" : int(census[1][10])/int(census[1][9])  
}
print(mapping_rel)

{'Asian/Pacific Islander': '15159516674625', 'Black': '40250635', 'Native American/Native Alaskan': '3739506', 'Hispanic': '44618105', 'White': '197318956'}
{'Asian/Pacific Islander': 0.051285408374063694, 'Black': 0.13036831320943656, 'Native American/Native Alaskan': 0.01211193536341892, 'Hispanic': 0.14451416946469361, 'White': 0.6390989721768869}


In [44]:
# Display Deaths by 100,000 of each race

race_per_hundredk = {}

for race in race_count:
    race_per_hundredk[race] = (int(race_count[race]) / int(mapping[race]) * 100000) 
    
print(race_per_hundredk)

{'Asian/Pacific Islander': 8.746980714890115e-06, 'White': 33.56849303419181, 'Native American/Native Alaskan': 24.521955573811088, 'Black': 57.8773477735196, 'Hispanic': 20.220491210910907}


We can filter our results, and restrict them to the Homicide intent. This will tell us what the gun-related murder rate per 100000 people in each racial category is. In order to do this, we'll need to redo our work in generating race_counts, but only count rows where the intent was Homicide.

We can do this by first extracting the intent column, then using the enumerate() function to loop through each index and value in the race column. If the value in the same position in intents is Homicide, we'll count the value in the race column.

Finally, we'll use the mapping dictionary to convert from raw counts to rates.

In [50]:
#extract intents
intents = [row[3] for row in guns_wo_header]

print(intents[:15])

print(races[:15])

['Suicide', 'Suicide', 'Suicide', 'Suicide', 'Suicide', 'Suicide', 'Undetermined', 'Suicide', 'Accidental', 'Suicide', 'Suicide', 'Suicide', 'Suicide', 'Suicide', 'Homicide']
['Asian/Pacific Islander', 'White', 'White', 'White', 'White', 'Native American/Native Alaskan', 'White', 'Native American/Native Alaskan', 'White', 'Black', 'White', 'Native American/Native Alaskan', 'White', 'Native American/Native Alaskan', 'White']


In [58]:
homicide_race_counts = {}

for i,race in enumerate(races):
    if intents[i] == "Homicide":
        if race in homicide_race_counts:
            homicide_race_counts[race] += 1
        else:
            homicide_race_counts[race] = 1
            
print(homicide_race_counts)

{'White': 9147, 'Asian/Pacific Islander': 559, 'Black': 19510, 'Native American/Native Alaskan': 326, 'Hispanic': 5634}


In [59]:
# Display **Homicide** Deaths by 100,000 of each race

homicide_race_per_hundredk = {}

for race in homicide_race_counts:
    homicide_race_per_hundredk[race] = (int(homicide_race_counts[race]) / int(mapping[race]) * 100000) 
    
print(homicide_race_per_hundredk)

{'White': 4.6356417981453335, 'Asian/Pacific Islander': 3.687452654316421e-06, 'Black': 48.471284987180944, 'Native American/Native Alaskan': 8.717729026240365, 'Hispanic': 12.627161104219914}


# Findings
* **Black** are leading with 48.5 per 100,000 capita in Homicide Detahs, followed by **Hispanics with 12.6%** and **Native Americans with 8.7%**

# More to explore:
* Figure out the link, if any, between month and homicide rate.
* Explore the homicide rate by gender.
* Explore the rates of other intents, like Accidental, by gender and race.
* Find out if gun death rates correlate to location and education.